### PR TITLE
Style main page

### PR DESCRIPTION
**Pull request by justcute**
### **Task**

Show all available movies on the main page. Use React and add styles.

Reviewer: @nieisnuje

### CODE REVIEW

File: app/controllers/api/products_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Api
+  class ProductsController < ApplicationController
+    def top
+      render json: Product.all, image_size: [1000, 350]
```
Comment (nieisnuje) [inline app/controllers/api/products_controller.rb:4]: you do not follow REST
this method should be named `index`

File: app/controllers/api/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    end
+
+    def new_products
+      render json: Product.last(8), image_size: [400, 600]
```
Comment (nieisnuje) [inline app/controllers/api/products_controller.rb:8]: you can move this to separate controller
and also name the method following REST

File: app/javascript/components/NewProducts.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      this.setState({products: res});
+    });
+  }
+
+  componentDidMount () {
+    this.getProducts()
+  }
+
+  render() {
+    const { products } = this.state;
```
Comment (nieisnuje) [inline app/javascript/components/NewProducts.jsx:29]: you don't need this line

File: app/javascript/components/NewProducts.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    return (
+      <div>
+        <NewProductsRow products={Object.values(products)} />
```
Comment (nieisnuje) [inline app/javascript/components/NewProducts.jsx:33]: `this.state.products`

File: app/javascript/components/NewProducts.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        <div class="row">
+          <button type="button" class="btn btn-dark center-block">Show more</button>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default TopRated
```
Comment (nieisnuje) [inline app/javascript/components/NewProducts.jsx:42]: always add empty line at the end of the file

File: app/javascript/components/NewProducts.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import NewProductsRow from './NewProductsRow';
+
+class TopRated extends React.Component {
```
Comment (nieisnuje) [inline app/javascript/components/NewProducts.jsx:4]: install Prettier extension to your ide and format all your js files with it

File: app/javascript/components/NewProductsRow.jsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import NonEditableProduct from './NonEditableNewProduct'
+
+const NewProductRow = ({ products }) => (
```
Comment (nieisnuje) [inline app/javascript/components/NewProductsRow.jsx:5]: why some of the components are classes and others are consts?

File: app/javascript/components/NonEditableCarouselProduct.jsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import propTypes from 'prop-types';
+
+const NonEditableCarouselProduct = ({ product, productKey }) => {
+  
+  if(product.image == null){
```
Comment (nieisnuje) [inline app/javascript/components/NonEditableCarouselProduct.jsx:6]: move this logic to a method

File: app/javascript/components/NonEditableNewProduct.jsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import propTypes from 'prop-types';
+
+const NonEditableNewProduct = ({ product }) => {
+  
+  if(product.image == null){
```
Comment (nieisnuje) [inline app/javascript/components/NonEditableNewProduct.jsx:6]: move to a method

File: app/javascript/components/NonEditableCarouselProduct.jsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    product.image = "https://hitsk.in/t/19/91/19/i_logo.gif";
```
Comment (nieisnuje) [inline app/javascript/components/NonEditableCarouselProduct.jsx:7]: move all urls to constants

File: app/javascript/components/TopRated.jsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+      this.setState({products: res});
+    });
+  }
+
+  componentDidMount () {
+    this.getProducts()
+  }
+
+  render() {
+    const { products } = this.state;
```
Comment (nieisnuje) [inline app/javascript/components/TopRated.jsx:29]: same as above
you can use `this.state.products`

File: app/serializers/product_serializer.rb [hunk 12]
```diff
@@ -0,0 +1,18 @@
+class ProductSerializer < ActiveModel::Serializer
+  include Rails.application.routes.url_helpers
```
Comment (nieisnuje) [inline app/serializers/product_serializer.rb:2]: why do you need this line?

File: app/javascript/components/NonEditableNewProduct.jsx [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  return(
+    <React.Fragment >
+      <div className="col-md-6">
+        <a href={productLink} className="btn btn-secondary">
+          <img src={product.image} alt="Product image" className="w-100"/>
+          <div className="card-body">
+            <p className="card-text">{product.title}</p>
+            <p>{product.description}</p>
+            <div className="text-muted">
+              <p><time>{product.updated_at.toLocaleString()}</time></p>
```
Comment (nieisnuje) [inline app/javascript/components/NonEditableNewProduct.jsx:20]: why do you need `toLocaleString`?

File: app/serializers/product_serializer.rb [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+  attributes :id, :title, :description, :trailer, :date, :duration, :sertification, :country, :updated_at,
+             :image, :type
+
+  def image
+    variant = object.image.variant(resize_to_fill: @instance_options[:image_size])
+    return rails_representation_url(variant, only_path: true) if variant
```
Comment (nieisnuje) [inline app/serializers/product_serializer.rb:8]: we don't need to use `return` in Ruby

File: app/controllers/api/products_controller.rb (hunk 2 above)
Comment (nieisnuje) [inline app/controllers/api/products_controller.rb:8]: why `.last(8)` ?
pass this number from frontend

File: app/serializers/product_serializer.rb (hunk 12 above)
Comment (justcute) [reply app/serializers/product_serializer.rb:2]: to get imagies

**Pull request closed without merging**